pkg/queue/prometheus_stats_reporter_test.go

```diff
@@ -34,73 +34,21 @@ const (
 	pod       = "helloworld-go-00001-deployment-8ff587cc9-7g9gc"
 )
 
-func TestNewPrometheusStatsReporter_negative(t *testing.T) {
-	tests := []struct {
-		name      string
-		errorMsg  string
-		result    error
-		namespace string
-		config    string
-		revision  string
-		pod       string
-	}{{
-		"Empty_Namespace_Value",
-		"Expected namespace empty error",
-		errors.New("namespace must not be empty"),
-		"",
-		config,
-		revision,
-		pod,
-	}, {
-		"Empty_Config_Value",
-		"Expected config empty error",
-		errors.New("config must not be empty"),
-		namespace,
-		"",
-		revision,
-		pod,
-	}, {
-		"Empty_Revision_Value",
-		"Expected revision empty error",
-		errors.New("revision must not be empty"),
-		namespace,
-		config,
-		"",
-		pod,
-	}, {
-		"Empty_Pod_Value",
-		"Expected pod empty error",
-		errors.New("pod must not be empty"),
-		namespace,
-		config,
-		revision,
-		"",
-	}}
-
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			if _, err := NewPrometheusStatsReporter(test.namespace, test.config, test.revision, test.pod, 1*time.Second); err.Error() != test.result.Error() {
-				t.Errorf("Got error msg from NewPrometheusStatsReporter(): '%+v', wanted '%+v'", err, test.errorMsg)
-			}
-		})
-	}
+type ReporterTestInput struct {
+	name                        string
+	reportingPeriod             time.Duration
+	concurrency                 float64
+	proxiedConcurrency          float64
+	reqCount                    float64
+	proxiedReqCount             float64
+	expectedReqCount            float64
+	expectedProxiedRequestCount float64
+	expectedConcurrency         float64
+	expectedProxiedConcurrency  float64
 }
 
-func TestReporterReport(t *testing.T) {
-	tests := []struct {
-		name            string
-		reportingPeriod time.Duration
-
-		concurrency        float64
-		proxiedConcurrency float64
-		reqCount           float64
-		proxiedReqCount    float64
-
-		expectedReqCount            float64
-		expectedProxiedRequestCount float64
-		expectedConcurrency         float64
-		expectedProxiedConcurrency  float64
-	}{{
+func GetReporterReportTests() []ReporterTestInput {
```

Comment:
No need to wrap them in a function I think. You could also just do:

```
var testCases := []struct { ... }
```

and then use that list in both functions.

Reply:
@markusthoemmes I thought both of the options you listed above, but it depends on code practices, I added the comment to clarify what it should run. Anyway I will copy the test cases as it is the easier although more verbose option.